x-pack/plugins/security/public/management/roles/edit_role/privileges/kibana/feature_table/feature_table.tsx

```diff
@@ -227,6 +234,17 @@ export class FeatureTable extends Component<Props, {}> {
               arrowDisplay={
                 canCustomizeSubFeaturePrivileges && hasSubFeaturePrivileges ? 'left' : 'none'
               }
+              onToggle={(isOpen: boolean) => {
+                if (isOpen) {
+                  this.state.expandedPrivilegeControls.add(feature.id);
+                } else {
+                  this.state.expandedPrivilegeControls.delete(feature.id);
+                }
+
+                this.setState({
+                  expandedPrivilegeControls: this.state.expandedPrivilegeControls,
```

Comment:
I don't think this is what we want. From React's perspective, we are setting a value equal to itself, so this could be considered a no-op. I think Oleg's original suggestion is the way to go, where we construct a new Set:

```ts
this.setState({
  expandedPrivilegeControls: new Set([...this.state.expandedPrivilegeControls]),
});
```

Reply:
I caught up with Oleg offline -- if this appears to be working as-is, then feel free to ignore this comment. Whatever y'all decide is fine with me!